packages/@blockprotocol/graph/src/react.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  RefObject,
+  useLayoutEffect,
+  useState,
+  VoidFunctionComponent,
+} from "react";
+
+import {
+  BlockGraphProperties,
+  GraphBlockHandler,
+  GraphEmbedderHandler,
+} from "./index";
+
+export type BlockComponent<
+  Properties extends Record<string, unknown> | null = null,
+> = VoidFunctionComponent<BlockGraphProperties<Properties>>;
+
+export const useGraphBlockService = ({
+  callbacks,
+  ref,
+}: {
+  callbacks: ConstructorParameters<typeof GraphBlockHandler>[0]["callbacks"];
+  ref: RefObject<HTMLElement>;
+}) => {
+  const [graphService, setGraphService] = useState<GraphBlockHandler>();
+  useLayoutEffect(() => {
+    if (ref.current) {
+      setGraphService(
+        new GraphBlockHandler({ callbacks, element: ref.current }),
+      );
+    }
+  }, [callbacks, ref]);
```

Comment:
Discussing on Slack the bug here, where the graph service isn't reconstructed when the element changes

Reply:
Thanks very much for the discussion - improved [here](https://github.com/blockprotocol/blockprotocol/pull/323/files/17500ecacb229bb60522a83b9f450b2aca10f260..9b77ba16366e1e528546cb4f18c666007e25980a)